packages/core/src/codewhisperer/commands/basicCommands.ts

```diff
@@ -326,3 +328,79 @@ export const signoutCodeWhisperer = Commands.declare(
         return auth.secondaryAuth.deleteConnection()
     }
 )
+
+export const registerToolkitApiCallback = Commands.declare(
+    { id: 'aws.amazonq.refreshConnectionCallback' },
+    () => async () => {
+        // While the Q/CW exposes an API for the Toolkit to register callbacks on auth changes,
+        // we need to do it manually here because the Toolkit would have been unable to call
+        // this API if the Q/CW extension started afterwards (and this code block is running).
+        if (isExtensionInstalled(VSCODE_EXTENSION_ID.awstoolkit)) {
+            const auth = Auth.instance
+            const toolkitExt = vscode.extensions.getExtension(VSCODE_EXTENSION_ID.awstoolkit)
+            const toolkitApi = toolkitExt?.exports
+            getLogger().info(`Trying to register toolkit callback. Toolkit is installed.`)
+            const registerCallback = once(async () => {
```

Comment:
not 100% sure but `once()` may have to be in an outer scope. It looks like it is getting created each time the command is executed, which would make a new instance of a one-off function each time... which means it runs multiple times.